sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseService.java

```diff
@@ -880,62 +888,119 @@ public Predicate createDefaultFilter(CriteriaBuilder cb, From<?, Case> root) {
-		for (Task task : tasks) {
-			taskService.deletePermanent(task);
-		}
+		// Delete surveillance reports related to this case
+		surveillanceReportService.getByCaseUuids(Collections.singletonList(caze.getUuid()))
+			.forEach(s -> surveillanceReportService.deletePermanent(s));
 
-		// Delete all prescriptions/treatments/clinical visits
+		// Delete documents related to this case
+		documentService.getRelatedToEntity(DocumentRelatedEntityType.CASE, caze.getUuid()).forEach(d -> documentService.markAsDeleted(d));
+
+		// Delete clinical management data
 		if (caze.getTherapy() != null) {
 			TherapyReferenceDto therapy = new TherapyReferenceDto(caze.getTherapy().getUuid());
-			treatmentService.findBy(new TreatmentCriteria().therapy(therapy)).stream().forEach(t -> treatmentService.deletePermanent(t));
-			prescriptionService.findBy(new PrescriptionCriteria().therapy(therapy)).stream().forEach(p -> prescriptionService.deletePermanent(p));
+			List<Treatment> treatments = treatmentService.findBy(new TreatmentCriteria().therapy(therapy));
+			treatments.forEach(t -> treatmentService.deletePermanent(t));
+			prescriptionService.findBy(new PrescriptionCriteria().therapy(therapy)).forEach(p -> prescriptionService.deletePermanent(p));
 		}
+
 		if (caze.getClinicalCourse() != null) {
 			ClinicalCourseReferenceDto clinicalCourse = new ClinicalCourseReferenceDto(caze.getClinicalCourse().getUuid());
-			clinicalVisitService.findBy(new ClinicalVisitCriteria().clinicalCourse(clinicalCourse))
-				.stream()
-				.forEach(c -> clinicalVisitService.deletePermanent(c));
+			List<ClinicalVisit> cvs = clinicalVisitService.findBy(new ClinicalVisitCriteria().clinicalCourse(clinicalCourse));
+			cvs.forEach(c -> clinicalVisitService.deletePermanent(c));
 		}
 
-		// Remove all events linked to case by removing the case_id from event participant
-		caze.getEventParticipants().stream().forEach(eventParticipant -> eventParticipant.setResultingCase(null));
+		// Remove the case from any S2S share info referencing it
+		sormasToSormasShareInfoService.getByAssociatedEntity(SormasToSormasShareInfo.CAZE, caze.getUuid()).forEach(s -> {
+			s.setCaze(null);
+			sormasToSormasShareInfoService.ensurePersisted(s);
+		});
 
-		// Unlink TravelEntries where this case is set as the resulting case
-		List<TravelEntry> travelEntries = travelEntryService.getAllByResultingCase(caze);
-		for (TravelEntry travelEntry : travelEntries) {
-			travelEntry.setResultingCase(null);
-			travelEntryService.ensurePersisted(travelEntry);
-		}
+		// Remove the case from any external share info referencing it
+		externalShareInfoService.getShareInfoByCase(caze.getUuid()).forEach(e -> {
+			e.setCaze(null);
+			externalShareInfoService.ensurePersisted(e);
+		});
 
-		// Unlink Immunizations where this case is set as the related case
-		immunizationService.unlinkRelatedCase(caze);
+		// Remove the case from its S2S origin info
+		Optional.ofNullable(caze.getSormasToSormasOriginInfo()).ifPresent(o -> {
```

Comment:
there is no join table between origin info so there is no need to do `o.getCases().remove(caze);` the origin server should be notified that the case has been deleted otherwise it would try to sync future changes

Reply:
Unsubscribing from the source system is currently out of scope (#7736).